src/networking.c

```diff
@@ -3876,7 +3881,37 @@ void unblockPostponedClients() {
     }
 }
 
-/* Pause clients up to the specified unixtime (in ms) for a given type of
+/* As pause-client feature refactored internally to become pause-actions,
+ * this is a wrapper that keeps pause-client backward compatible logic:
+ * 1. Keep higher end-time value between configured and the new one
+ * 2. Keep most restrictive action between configured and the new one
+ *
+ * Note: this logic is problematic to follow when pause/unpause actions.
+ * To the rest of action's purposes we take simplified approach (common
+ * implementation of pause-action) such that for a given action-purpose,
+ * new pause is the only one to count, irrelevant what was configured
+ * before */
```

Comment:
i think this note may also be redundant..
it was (arguably) problematic to follow, and you extracted it to a separate function,IMO there no need to describe the alternative that isn't implemented..